data/firestore/src/main/java/io/github/droidkaigi/confsched2020/data/firestore/internal/FirestoreImpl.kt

```diff
@@ -83,6 +120,40 @@ internal class FirestoreImpl @Inject constructor() : Firestore {
         firebaseAuth.signInAnonymously().await()
         Timber.debug { "signInIfNeeded end" }
     }
+
+    private fun getThumbsUpCounterRef(sessionId: SessionId): CollectionReference {
+        return FirebaseFirestore
+            .getInstance()
+            .collection("confsched/2020/sessions/${sessionId.id}/thumbsup_counters")
+    }
+
+    private suspend fun createShardsIfNeeded(counterRef: CollectionReference) {
+        if (!counterRef.fastGet().isEmpty) {
+            Timber.debug { "createShardsIfNeeded shards already exist" }
+            return
+        }
+
+        val tasks = arrayListOf<Task<Void>>()
+        (0 until NUM_SHARDS).forEach {
+            val makeShard = counterRef
+                .document(it.toString())
+                .set(mapOf(SHARDS_COUNT_KEY to 0))
+            tasks.add(makeShard)
+        }
+
+        try {
+            Tasks.whenAll(tasks).await()
+        } catch(e: Exception) {
+            println("⭐" + e.message)
```

Comment:
Umm...
I still catch the exception like this.
```
2020-02-02 19:09:35.472 7815-7815/io.github.droidkaigi.confsched2020.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.github.droidkaigi.confsched2020.debug, PID: 7815
    com.google.firebase.firestore.FirebaseFirestoreException: PERMISSION_DENIED: Missing or insufficient permissions.
        at com.google.firebase.firestore.util.Util.exceptionFromStatus(com.google.firebase:firebase-firestore@@20.2.0:121)
```

Reply:
👀

Reply:
Fixed by bf57b49
The cause was `isEmpty`.
https://github.com/DroidKaigi/conference-app-2020/pull/626#issuecomment-581157454